core/src/main/java/ai/timefold/solver/core/api/score/analysis/ConstraintAnalysis.java

```diff
@@ -156,6 +170,51 @@ public String constraintName() {
         return constraintRef.constraintName();
     }
 
+    /**
+     * Returns a diagnostic text that explains part of the score quality through the {@link ConstraintAnalysis} API.
+     *
+     * @return never null
+     */
+    public String summarize() {
+        StringBuilder summary = new StringBuilder();
+        summary.append("""
+                Explanation of score (%s):
+                    Constraint matches:
+                """.formatted(score));
+        Comparator<MatchAnalysis<Score_>> matchScoreComparator = comparing(MatchAnalysis::score);
+
+        var constraintMatches = matches();
+        if (constraintMatches == null) {
+            throw new IllegalArgumentException("""
+                    The constraint matches must be non-null.
+                    Maybe use ScoreAnalysisFetchPolicy.FETCH_ALL to request the score analysis
+                    """);
+        }
+        if (constraintMatches.isEmpty()) {
+            summary.append("""
+                            %s: constraint (%s) has no matches.
+                    """.formatted(score().toShortString(), constraintRef().constraintName()));
```

Comment:
Arguably, for short single-line strings, ```"""``` is unnecessary and more verbose.
But careful to maintain the newline at the end.